cpp/src/IceGrid/AdapterCache.cpp

```diff
@@ -704,16 +694,18 @@ ReplicaGroupEntry::getLocatorAdapterInfo(LocatorAdapterInfoSeq& adapters, int& n
         {
             //
             // This must be done outside the synchronization block since
-            // the trasnform() might call and lock each server adapter
+            // the transform() might call and lock each server adapter
             // entry. We also can't sort directly as the load of each
             // server adapter is not stable so we first take a snapshot of
             // each adapter and sort the snapshot.
             //
-            vector<pair<float, ServerAdapterEntryPtr> > rl;
-            transform(replicas.begin(), replicas.end(), back_inserter(rl), TransformToReplicaLoad(loadSample));
-            sort(rl.begin(), rl.end(), ReplicaLoadComp());
+            vector<pair<float, shared_ptr<ServerAdapterEntry>>> rl;
+            transform(replicas.begin(), replicas.end(), back_inserter(rl), bind(transformToReplicaLoad,
+                                                                                loadSample,
+                                                                                placeholders::_1));
+            sort(rl.begin(), rl.end(), replicaLoadComp);
             replicas.clear();
-            transform(rl.begin(), rl.end(), back_inserter(replicas), TransformToReplica());
+            transform(rl.begin(), rl.end(), back_inserter(replicas), transformToReplica);
```

Comment:
Wondering if it would be more readable to pass lambadas directly here

```
transform(replicas.begin(), replicas.end(), back_inserter(rl), 
    [loadSample] (const shared_ptr<ServerAdapterEntry>& value)
    {
        return { value->getLeastLoadedNodeLoad(loadSample), value }
    });
sort(rl.begin(), rl.end(), 
     [](const pair<float, std::shared_ptr<ServerAdapterEntry>>& lhs,
        const pair<float, std::shared_ptr<ServerAdapterEntry>>& rhs)
    {
        return lhs.first < rhs.first;
    }));
replicas.clear();
transform(rl.begin(), rl.end(), back_inserter(replicas), 
    [](const pair<float, shared_ptr<ServerAdapterEntry>>& value) 
    {
        return value->second;
    });
```